plugins/modules/net_tools/gandi_livedns.py

```diff
@@ -0,0 +1,194 @@
+    type: str
+    default: '@'
+    aliases: [ name ]
+  state:
+    description:
+    - Whether the record(s) should exist or not.
+    type: str
+    choices: [ absent, present ]
+    default: present
+  ttl:
+    description:
+    - The TTL to give the new record.
+    - Default is C(10800) when C(state=present).
+    type: int
+  type:
+    description:
+      - The type of DNS record to create.
+    type: str
+    required: true
+  values:
+    description:
+    - The record values.
+    - Required for C(state=present).
+    type: list
+    elements: str
+    aliases: [ content ]
+  domain:
+    description:
+    - The name of the Domain to work with (e.g. "example.com").
+    required: true
+    type: str
+'''
+
+EXAMPLES = r'''
+- name: Create a test A record to point to 127.0.0.1 in the my.com domain
+  gandi_livedns:
+    domain: my.com
+    record: test
+    type: A
+    values:
+    - 127.0.0.1
+    api_key: dummyapitoken
+  register: record
+
+- name: Create a mail CNAME record to www.my.com domain
+  gandi_livedns:
+    domain: my.com
+    type: CNAME
+    record: mail
+    values:
+    - www
+    api_key: dummyapitoken
+    state: present
+
+- name: Change its TTL
+  gandi_livedns:
+    domain: my.com
+    type: CNAME
+    values:
+    - www
+    ttl: 10800
+    api_key: dummyapitoken
+    state: present
+
+- name: Delete the record
+  gandi_livedns:
+    domain: my.com
+    type: CNAME
+    record: mail
+    api_key: dummyapitoken
+    state: absent
+'''
+
+RETURN = r'''
+record:
+    description: A dictionary containing the record data.
+    returned: success, except on record deletion
+    type: complex
+    contains:
+        values:
+            description: The record content (details depend on record type).
+            returned: success
+            type: list
+            sample:
+            - 192.0.2.91
+            - 192.0.2.92
+        record:
+            description: The record name.
+            returned: success
+            type: str
+            sample: www
+        ttl:
+            description: The time-to-live for the record.
+            returned: success
+            type: int
+            sample: 300
+        type:
+            description: The record type.
+            returned: success
+            type: str
+            sample: A
+        domain:
+            description: The domain associated with the record.
+            returned: success
+            type: str
+            sample: my.com
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.gandi_livedns_api import GandiLiveDNSAPI
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            api_key=dict(type='str', required=True, no_log=True),
+            record=dict(type='str', default='@', aliases=['name']),
+            state=dict(type='str', default='present', choices=['absent', 'present']),
+            ttl=dict(type='int', default=None),
```

Comment:
Please do not use `default=None`, as the sanity checkers assume that there is a non-null value when cross checking with documentation, which eventually force you to add more boilerplate to the docs block.

Reply:
Are you sure that happens? I can't reproduce that behavior with ansibe-test. (I do agree it's better to leave it away though. Makes it easier to read the argspec.)

Reply:
Maybe not in this particular case, but if you have ``default=None, required=True``, it triggers tgat validation for required and default at the same time, even though default is None.

Reply:
We can set a default value to 3600 or 7200, which looks like sane default for DNS entries.

Reply:
@russoz yep, for `required=True` it actually checks whether `default` is present. For other checks, it checks whether it is not `None` (if present).
@gthiemonge I think it's better to not have a default, if this is not needed for a very good reason. I've seen to defaults which didn't sound bad at some point turn into monsters which were really hard to get rid of somewhen in the future, so I think not adding some is usually better :)

Reply:
I've just remembered that I set a default value for ttl when state=present on line 166 (I think we don't need to specify the ttl when deleting a DNS entry)
What would be a better solution? Do you think we should add required=True and remove default=None?

Reply:
Adding `required=True` would also require it when `state=absent`, which isn't great for users. How about using `required_if` (see https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#argument-spec-dependencies) to make it only required when `state=present`?

Reply:
is it fixed?

Reply:
No idea on then, but it is now.